kernel/libs/aster-bigtcp/src/socket/bound/tcp_conn.rs

```diff
@@ -103,15 +188,20 @@ impl<E: Ext> RawTcpSocketExt<E> {
     /// dead if it is not closed.
     fn check_dead(&self, this: &Arc<TcpConnectionBg<E>>) -> TcpConnBecameDead {
         // FIXME: This is a temporary workaround to mark TimeWait socket as dead.
-        if self.state() == smoltcp::socket::tcp::State::Closed
-            || self.state() == smoltcp::socket::tcp::State::TimeWait
-        {
+        if self.state() == State::TimeWait {
+            return TcpConnBecameDead::TRUE;
+        }
+
+        // According to the current smoltcp implementation, a socket in the CLOSED state with the
+        // remote endpoint set means that an outgoing RST packet is pending. We cannot simply mark
+        // such a socket as dead.
+        if self.state() == State::Closed && self.remote_endpoint().is_none() {
```

Comment:
If the socket state is closed, and the remote endpoint is `None`, the socket won't be marked as dead. But when will the socket become dead? The socket state won't change any more, so `check_state` won't be called on this socket any more.